### PR TITLE
Fixes #32459 - remove pulp2 applicability

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
This still needs:
- [ ] testing
- [ ] test fixing

File: app/models/katello/concerns/pulp_database_unit.rb [hunk 1]
```diff
@@ -62,6 +62,48 @@ def immutable_unit_types
...
+        if hosts
+          hosts = ::Host.where(id: hosts) if hosts.is_a?(Array)
+          facet_repos = facet_repos.merge(hosts).reorder(nil)
+          facet_content_units = facet_content_units.merge(hosts).reorder(nil)
+        end
+
+        self.joins(repository_association_units).
+            where(repository_association_class.table_name => {:repository_id => facet_repos,
+                                                              content_unit_association_id => facet_content_units}).distinct
+      end
```
Comment (jlsherrill) [inline app/models/katello/concerns/pulp_database_unit.rb:105]: all of this was pulled from the old applicable_content_helper

File: app/models/katello/host/content_facet.rb [hunk 2]
```diff
@@ -97,6 +82,15 @@ def errata_counts
...
       end
 
+      def self.trigger_applicability_generation(host_ids)
+        host_ids = [host_ids] unless host_ids.is_a?(Array)
+
+        host_ids.each do |host_id|
+          ::Katello::ApplicableHostQueue.push_host(host_id)
+        end
+        ::Katello::EventQueue.push_event(::Katello::Events::GenerateHostApplicability::EVENT_TYPE, 0)
+      end
```
Comment (jlsherrill) [inline app/models/katello/host/content_facet.rb:92]: I tried to consolidate all applicability generation to use this method

File: config/katello.yaml.example [hunk 3]
```diff
@@ -66,19 +66,6 @@
     :crane_ca_cert_file: /etc/pki/katello/certs/katello-server-ca.crt
     :allow_push: false  # default false if omitted
 
-# Pulp 2 can be preferred over Pulp 3 for the desired content type.
-# Default is false.
-#  :use_pulp_2_for_content_type:
```
Comment (jlsherrill) [inline config/katello.yaml.example:71]: TODO: actually remove usage of this

Comment (jlsherrill) [reply config/katello.yaml.example:71]: actually this should probably be another pr

**Review by jjeffers — APPROVED** [review]
Tested this with a content host and used downlevel packages to force errata indications. Partial package upgrades were accurately reflected in the errata counts in Katello per host.

**Pull request merged**